tests/test_integrated_channels/test_api/test_blackboard/test_views.py

```diff
@@ -0,0 +1,131 @@
+        self.user.is_superuser = True
+        self.user.save()
+
+        self.enterprise_customer = factories.EnterpriseCustomerFactory(uuid=FAKE_UUIDS[0])
+        self.enterprise_customer_user = factories.EnterpriseCustomerUserFactory(
+            enterprise_customer=self.enterprise_customer,
+            user_id=self.user.id,
+        )
+
+        self.enterprise_customer_conf = factories.BlackboardEnterpriseCustomerConfigurationFactory(
+            enterprise_customer=self.enterprise_customer,
+        )
+
+    @mock.patch('enterprise.rules.crum.get_current_request')
+    def test_list(self, mock_current_request):
+        mock_current_request.return_value = self.get_request_with_jwt_cookie(
+            system_wide_role=ENTERPRISE_ADMIN_ROLE,
+        )
+        url = reverse('api:v1:blackboard:configuration-list')
+        response = self.client.get(url)
+        data = json.loads(response.content.decode('utf-8')).get('results')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(data), 1)
+        self.assertEqual(data[0].get('client_id'),
+                         str(self.enterprise_customer_conf.client_id))
+        self.assertEqual(data[0].get('client_secret'),
+                         str(self.enterprise_customer_conf.client_secret))
+        self.assertEqual(data[0].get('refresh_token'), '')
+        self.assertEqual(data[0].get('enterprise_customer'),
+                         self.enterprise_customer_conf.enterprise_customer.uuid)
+
+    @mock.patch('enterprise.rules.crum.get_current_request')
+    def test_get(self, mock_current_request):
+        mock_current_request.return_value = self.get_request_with_jwt_cookie(
+            system_wide_role=ENTERPRISE_ADMIN_ROLE,
+        )
+        url = reverse('api:v1:blackboard:configuration-detail', args=[self.enterprise_customer_conf.id])
+        response = self.client.get(url)
+        data = json.loads(response.content.decode('utf-8'))
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(data.get('blackboard_base_url'),
+                         self.enterprise_customer_conf.blackboard_base_url)
+        self.assertEqual(data.get('client_id'),
+                         str(self.enterprise_customer_conf.client_id))
+        self.assertEqual(data.get('client_secret'),
+                         str(self.enterprise_customer_conf.client_secret))
+        self.assertEqual(data.get('enterprise_customer'),
+                         self.enterprise_customer_conf.enterprise_customer.uuid)
+
+    @mock.patch('enterprise.rules.crum.get_current_request')
+    def test_post(self, mock_current_request):
+        self.enterprise_customer_conf.delete()
+        mock_current_request.return_value = self.get_request_with_jwt_cookie(
+            system_wide_role=ENTERPRISE_ADMIN_ROLE,
+        )
+        url = reverse('api:v1:blackboard:configuration-list')
+        payload = {
```

Comment:
Looking at the other tests, I noticed that we're not requiring a base url here. That seems fairly important for an integration, is there any desire to make it a required param?

Reply:
If you don't include it, it should already throw an error no?
And the front end should already prevent form submission without it.

Reply:
Bare minimum require it on the frontend I'd say, backend up to you, could potentially have people want to create their object and come back and flush it out later before enabling? No strong opinions here

Reply:
ok, decided noop. Thanks!